root/edit/EditData.js

```diff
@@ -0,0 +1,105 @@
+/*
+ * @flow strict-local
+ * Copyright (C) 2021 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import Layout from '../layout';
+import EditLink from '../static/scripts/common/components/EditLink';
+import EntityLink from '../static/scripts/common/components/EntityLink';
+import commaOnlyList from '../static/scripts/common/i18n/commaOnlyList';
+import {bracketedText} from '../static/scripts/common/utility/bracketed';
+import {formatPluralEntityTypeName}
+  from '../static/scripts/common/utility/formatEntityTypeName';
+import {getEditStatusName} from '../utility/edit';
+
+type Props = {
+  +edit: {...EditT, +id: number},
+  +relatedEntities: {[type: CoreEntityTypeT]: {
+    [entityId: string]: CoreEntityT,
+  }},
```

Comment:
Didn't know eslint accepted stuff like

```js
{foo: {
  1}}
```
but it looks kinda messy to me. I added some newlines, `$ReadOnly`, and some `+` sigils.

```suggestion
  +edit: $ReadOnly<{...EditT, +id: number}>,
  +relatedEntities: {
    +[type: CoreEntityTypeT]: {
      +[entityId: string]: CoreEntityT,
    },
  },
```

Reply:
Or actually, isn't that `GenericEditWithIdT` now?

Reply:
Yes, it is, I was making the change already but thanks for the reminder in case :)